pkg/clients/gcp/dns.go

```
@@ -147,10 +148,16 @@ func (c *gcpClient) DeleteAcmeChallengeResourceRecords(reqLogger logr.Logger, cr
 
 	// clean up records
 	err = c.deleteDnsRecords(zone, changes)
+
 	if err != nil {
-		return err
+		if e, ok := err.(*googleapi.Error); ok {
+			if e.Code == 400 && e.Message == "The 'entity.change' parameter is required but was missing." {
```

Ignoring this error means we aren't deleting these records.